c_glib/arrow-glib/decimal.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include <config.h>
+#endif
+
+#include <arrow-glib/decimal.hpp>
+
+G_BEGIN_DECLS
+
+/**
+ * SECTION: decimal128
+ * @title: Decimal128 class
+ * @include: arrow-glib/arrow-glib.h
+ *
+ * #GArrowDecimal128 is a decimal128 class.
+ *
+ * Since: 0.9.0
+ */
+
+typedef struct GArrowDecimal128Private_ {
+  std::shared_ptr<arrow::Decimal128> decimal128;
+} GArrowDecimal128Private;
+
+enum {
+  PROP_0,
+  PROP_DECIMAL128
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GArrowDecimal128,
+                           garrow_decimal128,
+                           G_TYPE_OBJECT)
+
+#define GARROW_DECIMAL128_GET_PRIVATE(obj)                 \
+  (G_TYPE_INSTANCE_GET_PRIVATE((obj),                   \
+                               GARROW_TYPE_DECIMAL128,     \
+                               GArrowDecimal128Private))
```

Comment:
Can you align trailing `\` as the following?

```diff
diff --git a/c_glib/arrow-glib/decimal.cpp b/c_glib/arrow-glib/decimal.cpp
index f5934cc9..9fe2b3fa 100644
--- a/c_glib/arrow-glib/decimal.cpp
+++ b/c_glib/arrow-glib/decimal.cpp
@@ -48,9 +48,9 @@ G_DEFINE_TYPE_WITH_PRIVATE(GArrowDecimal128,
                            garrow_decimal128,
                            G_TYPE_OBJECT)
 
-#define GARROW_DECIMAL128_GET_PRIVATE(obj)                 \
-  (G_TYPE_INSTANCE_GET_PRIVATE((obj),                   \
-                               GARROW_TYPE_DECIMAL128,     \
+#define GARROW_DECIMAL128_GET_PRIVATE(obj)                      \
+  (G_TYPE_INSTANCE_GET_PRIVATE((obj),                           \
+                               GARROW_TYPE_DECIMAL128,          \
                                GArrowDecimal128Private))
 
 static void
```

We can use `c-backslash-region` (`C-c C-\`) on Emacs for this but I don't know the same feature for Vim...

It may be a time that we introduce clang-format or something.